superset-frontend/src/dashboard/components/nativeFilters/FilterBar/index.tsx

```diff
@@ -187,8 +187,11 @@ const FilterBar: React.FC<FiltersBarProps> = ({
     setIsFilterSetChanged(tab !== TabIds.AllFilters);
     setDataMaskSelected(draft => {
       const children = cascadeChildren[filter.id] || [];
-      // force instant updating on initialization or for parent filters
-      if (filter.isInstant || children.length > 0) {
+      // force instant updating on initialization or for parent filters when dataMaskSelected has filter
+      if (
+        dataMaskSelected[filter.id] &&
```

Comment:
Ensure that DataMaskSelected contains the filter, rather than the initial object(empty object).